grunttasks/xssmitigate.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var xss = require('xss');
+
+module.exports = function (grunt) {
+  grunt.registerTask('xss-flag', 'Flag potentially harmful files', [
+    'clean',
+    'selectconfig:dist',
+    'l10n-generate-pages',
+    'l10n-json-to-html',
+    'xss-parse'
+    ]
+  );
+
+  grunt.registerMultiTask('xss-parse', 'Parse HTML files to check for XSS vectors', function () {
+    var templateRegex = /\%\(\w+\)s/;
+    this.files.forEach(function (file) {
+      // iterate through each HTML file in the .tmp folder
+      var src = file.src[0];
+      var contents = grunt.file.read(src);
+      // whitelist of allowed tags and attributes, nothing else is allowed
+      var options = {
+        whiteList: {
+          a: ['href', 'id'],
+          em: [],
+          span: ['id', 'tabindex'],
+          strong: []
+        },
+        onTagAttr: function (tag, name, value, isWhiteAttr) {
+          // On encountering any tag, check if the tag has a valid attribute.
+          // If the attribute is valid, check for its value, flag if illegal
+          // Illegal includes any mention of the string 'javascript', case insensitive
+          // Templated strings are allowed i.e strings of the form '%(termsURI)s'
+          // If these are considered to be not secure, we should change
+          if (isWhiteAttr && xss.safeAttrValue(tag, name, value) === '' && ! templateRegex.test(value)) {
+            grunt.fail.fatal('INVALID VALUE FOR ATTRIBUTE ' + tag + ' ' + name + ' ' + value + ' ' + src);
```

Comment:
Weird. It works for `grunt.log.writeln()`.

Workaround, use `util.format()` for interpolation:

```js
var msg = util.format('INVALID VALUE FOR ATTRIBUTE %s %s %s %s', 'a', 'b', 'c', 'd');
grunt.fail.fatal(msg); // Fatal error: INVALID VALUE FOR ATTRIBUTE a b c d
```